profit_analysis/prices.py

```diff
@@ -61,41 +62,52 @@ async def get_decimals_from_token(self, token):
         return decimals
 
     async def create(self, token_target_address, max_retries=24):
-        trials = 0
-        n_trials = 3
-        while trials < n_trials:
-            trials += 1
-            try:
-                self._token_target_address = token_target_address
-                if token_target_address != self._token_base_address:
-                    factory = self.w3_provider.w3_provider_archival.eth.contract(
-                        address=self._factory, abi=UNISWAP_V2_FACTORY_ABI
-                    )
-                    pair_address = await factory.functions.getPair(
-                        self._token_base_address, token_target_address
-                    ).call()
-
-                    pair_contract = self.w3_provider.w3_provider_archival.eth.contract(
-                        address=pair_address, abi=UNISWAP_V2_PAIR_ABI
-                    )
-
-                    self._pair = pair_contract
-                    self._token_base_decimals = (
-                        10
-                        ** await self.get_decimals_from_token(self._token_base_address)
-                    )
-                    self._token_target_decimals = (
-                        10 ** await self.get_decimals_from_token(token_target_address)
-                    )
-                    token_n = await self.is_target_token0_or_token1()
-                    self._is_target_token0_or_token1 = token_n
-                    self._max_retries = max_retries
-                return self
-            except Exception as e:
-                print(f"Error ({trials}/{n_trials}), retrying  create  -  {e}")
-                sleep(0.05)
-        W3.rotate_rpc_url()
-        return await self.create(token_target_address, max_retries)
+        if max_retries > 0:
+            trials = 0
+            n_trials = 3
+            while trials < n_trials:
+                trials += 1
+                try:
+                    self._token_target_address = token_target_address
+                    if token_target_address != self._token_base_address:
+                        factory = self.w3_provider.w3_provider_archival.eth.contract(
+                            address=self._factory, abi=UNISWAP_V2_FACTORY_ABI
+                        )
+                        pair_address = await factory.functions.getPair(
+                            self._token_base_address, token_target_address
+                        ).call()
+                        if pair_address == NULL_ADDRESS:
```

Comment:
It may happen that a token exists but there is no pool vs USDC, so we just skip it.